### PR TITLE
Add Vietnamese localisation

### PR DESCRIPTION
**Pull request by Kritiqual**
This is the Vietnamese translation. Hope it will be released!

### CODE REVIEW

File: aura/lib/Aura/Languages.hs [hunk 1]
```diff
@@ -496,6 +521,7 @@ missingPkg_4 pns = \case
   Dutch      -> vsep $ "The volgende afhankelijkheidscycli zijn gedetecteerd:" : pns'
   Ukrainian  -> vsep $ "Було помічено цикл залежностей:" : pns'
   Romanian   -> vsep $ "Aceste cicluri de dependență a fost detectate:" : pns'
+  Vietnamese -> vsep $ "Phát hiện chu kỳ gói phụ thuộc: " pns'
```
Comment (ratijas) [inline aura/lib/Aura/Languages.hs:524]: missing colon
```suggestion
  Vietnamese -> vsep $ "Phát hiện chu kỳ gói phụ thuộc: " : pns'
```

File: aura/lib/Aura/Languages.hs [hunk 2]
```diff
@@ -672,6 +705,7 @@ confirmIgnored_1 (bt . pnName -> p) = \case
     Esperanto  -> p <> " estas markita kiel malatenta. Ĉu instali?"
     Dutch      -> p <> " is gemarkeerd als genegeerd. Toch installeren?"
     Romanian   -> p <> " e marcat ca ignorat. Să se instaleze oricum?"
+    Vietnamese -> p <> "được đáng dấu là Bỏ qua. Vẫn cài đặt nó?"
```
Comment (ratijas) [inline aura/lib/Aura/Languages.hs:708]: missing space
```suggestion
    Vietnamese -> p <> " được đáng dấu là Bỏ qua. Vẫn cài đặt nó?"
```

File: aura/lib/Aura/Languages.hs [hunk 3]
```diff
@@ -807,6 +846,7 @@ reportPkgbuildDiffs_1 (bt . pnName -> p) = \case
     Dutch      -> p <> " heeft nog geen opgeslagen PKGBUILD."
     Ukrainian  -> "В " <> p <> " ще не зберігається PKGBUILD."
     Romanian   -> p <> " încă nu are un PKGBUILD descărcat."
+    Vietnamese -> p <> "không có sẵn PKGBUILD."
```
Comment (ratijas) [inline aura/lib/Aura/Languages.hs:849]: missing space
```suggestion
    Vietnamese -> p <> " không có sẵn PKGBUILD."
```

File: aura/lib/Aura/Languages.hs [hunk 4]
```diff
@@ -1008,6 +1056,7 @@ cleanStates_2 n@(bt . tshow -> s) = \case
     Dutch      -> s <> " pakketstatussen worden behouden. De rest verwijderen?"
     Ukrainian  -> s <> " стан пакунків будуть залишені. Видалити решту?"
     Romanian   -> "Stările pachetului " <> s <> " vor fi păstrate. Să se șteargă restul?"
+    Vietnamese -> "Trạng thái của gói" <> s <> "sẽ được lưu lại. Loại bỏ phần còn lại?"
```
Comment (ratijas) [inline aura/lib/Aura/Languages.hs:1059]: missing spaces
```suggestion
    Vietnamese -> "Trạng thái của gói " <> s <> " sẽ được lưu lại. Loại bỏ phần còn lại?"
```

File: aura/lib/Aura/Languages.hs [hunk 5]
```diff
@@ -1071,6 +1123,7 @@ cleanStates_6 n = \case
   Dutch     -> pretty n <+> "hiervan zijn vastgezet, en worden niet verwijderd."
   Ukrainian -> pretty n <+> "були закріплені та залишуться недоторканими."
   Romanian  -> pretty n <+> "dintre astea sunt fixate, și nu vor fi șterse."
+  Vietnamese -> pretty n <+> " trong số chúng đã được ghim, và sẽ không bị loại bỏ."
```
Comment (ratijas) [inline aura/lib/Aura/Languages.hs:1126]: space is not needed
```suggestion
  Vietnamese -> pretty n <+> "trong số chúng đã được ghim, và sẽ không bị loại bỏ."
```

File: aura/lib/Aura/Languages.hs [hunk 6]
```diff
@@ -1397,6 +1464,7 @@ cleanCache_7 (bt . tshow -> ps) (bt . tshow -> bytes) = \case
     Spanish    -> "La caché contiene " <> ps <> " paquetes, consumiendo " <> bytes <> " megabytes."
     Ukrainian  -> "Кеш містить " <> ps <> " пакунків, які використовують " <> bytes <> " МБ місця."
     Romanian   -> "Cache-ul conține " <> ps <> " pachete, consumând " <> bytes <> " MB."
+    Vietnamese -> "Có " <> ps <> " gói trong cache, chiếm " <> bytes <> "megabytes."
```
Comment (ratijas) [inline aura/lib/Aura/Languages.hs:1467]: missing space near the end
```suggestion
    Vietnamese -> "Có " <> ps <> " gói trong cache, chiếm " <> bytes <> " megabytes."
```

File: aura/lib/Aura/Languages.hs [hunk 7]
```diff
@@ -1782,6 +1870,7 @@ hotEdit_1 (bt . pnName -> p) = \case
     Dutch      -> "Wilt u het PKGBUILD-bestand van " <> p <> " bewerken?"
     Ukrainian  -> "Бажаєте відредагувати PKGBUILD для пакунку " <> p <> "?"
     Romanian   -> "Doriți să modificați PKGBUILD-ul pachetului " <> p <> "?"
+    Vietnamese -> "Bạn có muốn chỉnh sửa PKGBUILD của" <> p <> "?"
```
Comment (ratijas) [inline aura/lib/Aura/Languages.hs:1873]: missing space
```suggestion
    Vietnamese -> "Bạn có muốn chỉnh sửa PKGBUILD của " <> p <> "?"
```

File: aura/lib/Aura/Languages.hs [hunk 8]
```diff
@@ -510,6 +536,7 @@ missingPkg_5 (PkgName p) = \case
   Dutch     -> bt p <> " bestaat niet."
   Ukrainian -> "Пакунок " <> bt p <> " не існує."
   Romanian  -> "Pachetul " <> bt p <> " nu există."
+  Vietnamese -> bt p <> "không tồn tại."
```
Comment (ratijas) [inline aura/lib/Aura/Languages.hs:539]: missing space
```suggestion
  Vietnamese -> bt p <> " không tồn tại."
```

**Review by ratijas — CHANGES_REQUESTED** [review]
Good job! I left some comments, please take a look.

@fosskers those `<>` and `<+>` seem to confuse a lot of people, and are the primary source of mistakes for translators. That's noteworthy.

**Comment by fosskers** [comment]
> those <> and <+> seem to confuse a lot of people

Localisation in Aura 4 is _way_ simpler and won't suffer from that issue, so I'm going to leave it (the Haskell) as it is.

**Comment by fosskers** [comment]
Thank you! I did actually merge this after doing some rebasing, so your contribution has been included!

**Pull request closed without merging**